extras/dcgoss/dcgoss

```diff
@@ -10,26 +11,31 @@ info() {
 }
 error() {
     echo -e "ERROR: $*" >&2;
+    [[ -e "$service" ]] && docker logs "$service"
     exit 1;
 }
 
 cleanup() {
     set +e
     { kill "$log_pid" && wait "$log_pid"; } 2> /dev/null
-    rm -rf "$tmp_dir"
+    [ "$DEBUG" ] || rm -rf "$tmp_dir"
     if [[ -n "$service" ]]; then
-        info "Deleting container"
-        docker-compose down -v > /dev/null
+        #TODO setting for down -v vs stop?
+        #info "Deleting container"
+        #docker-compose down -v > /dev/null
+        info "Stopping container"
+        echo "Deleting $(docker rm -f "$service")"
+        docker-compose stop > /dev/null
     fi
 }
 
 run(){
     # Copy in goss
     cp "${GOSS_PATH}" "$tmp_dir/goss"
     chmod 755 "$tmp_dir/goss"
-    [[ -e "${GOSS_FILES_PATH}/goss.yaml" ]] && cp "${GOSS_FILES_PATH}/goss.yaml" "$tmp_dir"
-    [[ -e "${GOSS_FILES_PATH}/goss_wait.yaml" ]] && cp "${GOSS_FILES_PATH}/goss_wait.yaml" "$tmp_dir"
-    [[ -n "${GOSS_VARS}" ]] && [[ -e "${GOSS_FILES_PATH}/${GOSS_VARS}" ]] && cp "${GOSS_FILES_PATH}/${GOSS_VARS}" "$tmp_dir"
+    [[ -e "${GOSS_FILES_PATH}/${GOSS_FILE:-goss.yaml}" ]] && install -m ugo+rw "${GOSS_FILES_PATH}/${GOSS_FILE:-goss.yaml}" "$tmp_dir"
```

Comment:
As far as I remember yes. Too broad permissions do not impact security, as the file is only copied into the container for the duration of the test. In the end I had to set the permissions since the user I was running tests with was nobody.

https://github.com/aelsabbahy/goss/issues/493 is a related issue